test/runtests.py

```diff
@@ -7,43 +7,64 @@
 import sys
 import os
 import subprocess as SP
+import argparse, textwrap
+import xml.etree.ElementTree as ET
+
+parser = argparse.ArgumentParser(
+    description='ChakraCore *nix Test Script',
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=textwrap.dedent('''\
+        Samples:
+
+        test all folders:
+            {0}
+
+        test only Array:
+            {0} Array
+
+        test a single file:
+            {0} Basics/hello.js
+    '''.format(sys.argv[0]))
+    )
+parser.add_argument('folders', metavar='folder', nargs='*',
+                    help='folder subset to run tests')
+parser.add_argument('-b', '--binary', metavar='binary', help='ch full path');
+parser.add_argument('-d', '--debug', action='store_true',
+                    help='use debug build');
+parser.add_argument('-t', '--test', action='store_true', help='use test build');
+parser.add_argument('--x86', action='store_true', help='use x86 build');
+parser.add_argument('--x64', action='store_true', help='use x64 build');
+args = parser.parse_args()
 
-test_all = True
-test_root = os.path.dirname(os.path.realpath(__file__))
-
-# ugly trick
-ch_path = os.path.join(os.path.dirname(test_root), "BuildLinux/ch")
 
-if not os.path.isfile(ch_path):
-    print "BuildLinux/ch not found. Did you run ./build.sh already?"
+test_root = os.path.dirname(os.path.realpath(__file__))
+repo_root = os.path.dirname(test_root)
+
+# arch: x86, x64
+arch = 'x86' if args.x86 else ('x64' if args.x64 else None)
+if arch == None:
+    arch = os.environ.get('_BuildArch', 'x86')
+
+# flavor: debug, test, release
+flavor = 'debug' if args.debug else ('test' if args.test else None)
+if flavor == None:
+    flavor = {'chk':'debug', 'test':'test', 'fre':'release'}\
+                    [os.environ.get('_BuildType', 'fre')]
+
+# binary: full ch path
+binary = args.binary
+if binary == None:
+    if sys.platform == 'win32':
+        binary = os.path.join(repo_root,
+            'Build/VcBuild/bin/{}_{}/ch.exe'.format(arch, flavor))
+    else:
+        binary = os.path.join(repo_root, "BuildLinux/ch")
+if not os.path.isfile(binary):
+    print '{} not found. Did you run ./build.sh already?'.format(binary)
     sys.exit(1)
 
-if len(sys.argv) > 1:
-    if sys.argv[1] in ['-?', '--help']:
-        print "ChakraCore *nix Test Script\n"
-
-        print "Usage:"
-        print "test.py <optional test path>\n"
-
-        print "-?, --help    : Show help\n"
-
-        print "Samples:"
-        print "test only Array:"
-        print "\t./test.py Array\n"
-        
-        print "test a single file:"
-        print "\t./test.py Basics/hello.js\n"
-
-        print "test all folders:"
-        print "\t./test.py"
-        sys.exit(0)
-    test_all = None
-
-test_dirs=['']
-if test_all:
-    test_dirs = os.listdir(test_root)
-else:
-    test_dirs[0] = sys.argv[1]
+pass_count = 0
+fail_count = 0
```

Comment:
How about defining a class object for test results and keeping all the test related global information inside? The global variables are usually tend to mix-up at some point.